src/core/xlio_extra.h

```diff
@@ -37,6 +37,8 @@
 #include <stdint.h>
 #include <netinet/in.h>
 #include <sys/socket.h>
+#include <resolv.h>
```

Comment:
remove resolv.h -- look like it's not needed.

Reply:
It is needed for the __recv* functions

Reply:
you mean __recv_chk() and __recvfrom_chk()? resolv.h is not needed for them. It's purpose is to define `res_state` for __res_iclose() which should be removed from xlio_extra.h. I don't see other usage of resolv.h here.

Reply:
It would be shocking for SPDK or some other future client to statically link against XLIO and find out that we override the __res_iclose, and there is no way to opt out.

Reply:
OK